api/unstable/serialize.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <s2n.h>
+
+/**
+ * @file serialize.h
+ * 
+ * "Serialize" is a feature that takes an established s2n-tls connection object and "serializes" it
+ * into a transferable object to be sent off-box or stored. This transferable object can then
+ * be "deserialized" to create an s2n-tls connection object that can talk to the original peer with
+ * the same encryption keys.
+ *
+ * Note that this feature is dangerous because it provides cryptographic material from a TLS session
+ * in plaintext. Users MUST both encrypt and MAC the contents of the outputted material to provide
+ * secrecy and integrity if this material is transported off-box. DO NOT send this material off-box
+ * without a plan to do so.
+ */
+
+/* Indicates which serialized connection version will be provided. */
+typedef enum {
+    S2N_SERIALIZED_CONN_NONE = 0,
+    S2N_SERIALIZED_CONN_V1 = 1
+} s2n_serialization_version;
+
+/**
+ * Request a specific serialized connection version to be provided post-handshake.
+ *
```

Comment:
You probably need to explain that this MUST be called in order to be allowed to call serialize later. You probably don't need to go into the why, although maybe the why will come up if you explain how / when you'd bump the version.